### ISSUE TITLE
Parsing Indentations with “Tab” working incorrectly

### ISSUE THREAD

**Issue by VexSpectre** [GnaspGames/Smelt#41]
This is my original code:

```
#Init Scoreboards
{"type":"impulse", "auto":false}
/scoreboard     objectives add sHambre      food
    {"type":"chain", "auto":true}
    /scoreboard objectives add sHp          health
```

When I compile it, I get this in the command blocks:
1: `/scoreboard \tobjectives add sHambre\t\tfood`
2: `/scoreboard objectives add sHp \t\t\thealth`

As you can see there are some **_\t_** that are corrupting the commands.
I’m using Sublime Text 3 to write the files and I haven’t created or modified the config file for smelt.

**Comment by GnaspGames** [comment]
Thanks for the bug report. I suspect this has something to do with string escaping required to make sure that some of the JSON/NBT to properly parsed. I'll have a look at it.

**Comment by GnaspGames** [comment]
I'm having trouble replicating this issue on Windows. 

You're using Linux, is that correct?

**Comment by GnaspGames** [comment]
@VexSpectre ^

**Comment by GnaspGames** [comment]
Sorry, please excuse me. My editor was automatically replacing tabs with spaces.

**Comment by VexSpectre** [comment]
As 0.9.1, the command:

```
#Scoreboards
>{"type":"impulse-chain"}
    //Hambre
    /scoreboard     objectives add sHambre      food
```

Compiles into:

![untitled-1](https://cloud.githubusercontent.com/assets/20722837/17460205/f7f0c1b2-5c18-11e6-9109-98fc8066f5ff.png)

**Comment by GnaspGames** [comment]
@VexSpectre is this okay?

It was thought that the tab character should be replaced with 4 spaces since someone might use them in a say/tellraw for formatting. 

In your case it probably means that the arguments to the scoreboard command are incorrect. Personally I don't feel you should be using tabs like this.

**Comment by JBartelsson** [comment]
I agree with Gnasp here. If you write a command to your Minecraft Chat box/console you also get an error because of too many spaces

**Comment by VexSpectre** [comment]
@GnaspGames I use them for visual alignment with other commands on my sheet, it would be cool if the indentations outside of JSON blocks were deleted when parsed. I'm not using spaces, I can copy and paste the exact same command manually form Sublime / Notepad++ and have no errors, but thanks for the response and excuse me.

**Comment by GnaspGames** [comment]
@VexSpectre I think when you copy/paste from a text editor directly into the command block then the tabs must be stripped out. I suppose that's a good point; you can't actually format with tabs directly inside Minecraft. Formatting (even inside a JSON block) would have to be done with spaces.

Okay, I'm happy to rethink this. Tabs should be replaced with 1 space, and if formatting text inside MC is a problem for some, we will recommend they use spaces and not tabs.

**Comment by GnaspGames** [comment]
This will be fixed when eb4e8bf09495a12a1b2e2a0fb5c2f208a9c3295b is released.

**Comment by GnaspGames** [comment]
Closed with version 0.9.3
